sharktank/sharktank/layers/testing.py

```diff
@@ -373,4 +420,131 @@ def make_random_moe_block_theta(
             name=f"blk.{block_idx}.layer_output_norm.weight",
             data=make_rand_torch((in_dim), dtype=dtype_norm),
         )
+
+    return Theta(res)
+
+
+def make_fused_qkv_attention_block_theta(
+    *,
+    block_idx: int,
+    head_count: int,
+    head_count_kv: int,
+    head_dim: int,
+    embedding_length: int,
+    dtype: torch.dtype,
+    dtype_norm: torch.dtype,
+    with_bias: bool = True,
+    with_attention_sinks: bool = True,
+    with_attention_norm: bool = True,
+) -> Theta:
+    """Create theta for attention block with fused QKV projection.
+
+    Generates attention weights with a single fused QKV projection instead of
+    separate Q/K/V projections. Supports optional bias terms and attention sinks.
+
+    """
+    q_size = head_count * head_dim
+    k_size = head_count_kv * head_dim
+    v_size = head_count_kv * head_dim
+    qkv_size = q_size + k_size + v_size
+
+    res = {}
+
+    if with_attention_norm:
+        res["attn_norm.weight"] = DefaultPrimitiveTensor(
+            name=f"blk.{block_idx}.attn_norm.weight",
+            data=make_rand_torch((embedding_length,), dtype=dtype_norm),
+        )
+
+    res["attn.wqkv.weight"] = DefaultPrimitiveTensor(
+        name=f"blk.{block_idx}.attn.wqkv.weight",
+        data=make_rand_torch((qkv_size, embedding_length), dtype=dtype),
+    )
+
+    res["attn_output.weight"] = DefaultPrimitiveTensor(
+        name=f"blk.{block_idx}.attn_output.weight",
+        data=make_rand_torch((embedding_length, q_size), dtype=dtype),
+    )
+
+    if with_bias:
+        res["attn.wqkv.bias"] = DefaultPrimitiveTensor(
+            name=f"blk.{block_idx}.attn.wqkv.bias",
+            data=make_rand_torch((qkv_size,), dtype=dtype),
+        )
+        res["attn_output.bias"] = DefaultPrimitiveTensor(
+            name=f"blk.{block_idx}.attn_output.bias",
+            data=make_rand_torch((embedding_length,), dtype=dtype),
+        )
+
+    if with_attention_sinks:
+        res["attn_sinks"] = DefaultPrimitiveTensor(
+            name=f"blk.{block_idx}.attn_sinks",
+            data=make_rand_torch((head_count,), dtype=dtype),
+        )
+
+    return Theta(res)
+
+
+def make_moe_block_with_bias_theta(
```

Comment:
Why is this a separate function rather than making the existing `make_random_moe_block_theta` more generic?